lib/charms/opensearch/v0/models.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Cluster-related data structures / model classes."""
+import json
+from typing import List
+
+
+class Node:
+    """Data class representing a node in a cluster."""
+
+    def __init__(self, name: str, roles: List[str], ip: str):
+        self.name = name
+        self.roles = list(set(roles))
```

Comment:
I would probably tend to use the types that make most sense for the use case, adapting the serialization, rather than adapting the business logic. If the serialization into json does not work, it should also be fairly easy to extend the class (see [here](https://stackoverflow.com/questions/8230315/how-to-json-serialize-sets) for an example). I mean, the code required is comparable to the one needed for handling a set using list types.

Beside, when reading JSON you should not need to create a custom handler, if you use `Sequence[str]` for the argument in the constructor and then setting `self.roles=set(roles)`, thus making sure the attribute is indeed a set